contracts/Registrar.sol

```diff
@@ -6,86 +6,144 @@ import "@ensdomains/ens/contracts/ENS.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
 import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
 
+// commitments are kept in a seperate contract to allow the state to be reused
+// between different versions of the registrar
+contract Commitments {
+    address public owner;
+    modifier auth { require(msg.sender == owner, "Commitments: unauthorized"); _; }
+    event SetOwner(address usr);
+
+    /// Mapping from the commitment to the block number in which the commitment was made
+    mapping(bytes32 => uint) public commited;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function setOwner(address usr) external auth {
+        owner = usr;
+        emit SetOwner(usr);
+    }
+
+    function commit(bytes32 commitment) external auth {
+        commited[commitment] = block.number;
+    }
+
+    function mkCommitment(
+        string calldata name, address owner, uint salt
+    ) external pure returns (bytes32) {
+        return keccak256(abi.encodePacked(name, owner, salt));
+    }
+}
+
 contract Registrar {
-    /// The ENS registry.
-    ENS public immutable ens;
 
-    /// The price oracle.
-    address public oracleAddress;
+    // --- DATA ---
 
-    /// The Rad/Eth exchange.
-    address public exchangeAddress;
+    /// The ENS registry.
+    ENS public ens;
 
     /// The Radicle ERC20 token.
     ERC20Burnable public immutable rad;
 
+    /// The commitment storage contract
+    Commitments public immutable commitments = new Commitments();
+
+    /// The namehash of the `eth` TLD in the ENS registry, eg. namehash("eth").
+    bytes32 public constant ethNode = keccak256(abi.encodePacked(bytes32(0), keccak256("eth")));
+
     /// The namehash of the node in the `eth` TLD, eg. namehash("radicle.eth").
-    bytes32 public immutable domain;
+    bytes32 public constant radNode = keccak256(abi.encodePacked(ethNode, keccak256("radicle")));
 
     /// The token ID for the node in the `eth` TLD, eg. sha256("radicle").
-    uint256 public immutable tokenId;
+    uint256 public constant tokenId = uint(keccak256(abi.encodePacked("radicle")));
 
-    /// Registration fee in *USD*.
-    uint256 public registrationFeeUsd = 10;
+    /// The minimum number of blocks that must have passed between a commitment and name registration
+    // TODO: justify this as a default value...
+    uint256 public minCommitmentAge = 50;
```

Comment:
What is the purpose of having a minimum greater than `1`? Is it in case of a re-org? Perhaps worth documenting.

Reply:
yes, to cover network congestions, re-orgs, or to cover cases where the `register` tx was submitted with a very low gas fee.

Definately agree re: docs, was planning to write up a deeper analysis of the commit / reveal scheme in the audit report.